internal/gke/update.go

```diff
@@ -373,23 +373,30 @@ func UpdateLabels(
 	if config.Spec.Labels == nil || reflect.DeepEqual(config.Spec.Labels, upstreamSpec.Labels) || (upstreamSpec.Labels == nil && len(config.Spec.Labels) == 0) {
 		return NotChanged, nil
 	}
-	cluster, err := GetCluster(ctx, client, &config.Spec)
-	if err != nil {
-		return NotChanged, err
-	}
 	logrus.Infof("updating cluster labels for cluster [%s]", config.Name)
-	_, err = client.Projects.
-		Locations.
-		Clusters.
-		SetResourceLabels(
-			ClusterRRN(config.Spec.ProjectID, Location(config.Spec.Region, config.Spec.Zone), config.Spec.ClusterName),
-			&gkeapi.SetLabelsRequest{
-				LabelFingerprint: cluster.LabelFingerprint,
-				ResourceLabels:   config.Spec.Labels,
-			},
-		).Context(ctx).
-		Do()
-	if err != nil {
+	// Even though we're getting the up to date cluster here, sometimes the fingerprint isn't right and we need to try again with a fresh fingerprint.
+	for i := 0; i < 3; i++ {
+		cluster, err := GetCluster(ctx, client, &config.Spec)
+		if err != nil {
+			return NotChanged, err
+		}
+		_, err = client.Projects.
+			Locations.
+			Clusters.
+			SetResourceLabels(
+				ClusterRRN(config.Spec.ProjectID, Location(config.Spec.Region, config.Spec.Zone), config.Spec.ClusterName),
+				&gkeapi.SetLabelsRequest{
+					LabelFingerprint: cluster.LabelFingerprint,
+					ResourceLabels:   config.Spec.Labels,
+				},
+			).Context(ctx).
+			Do()
+		if err == nil {
+			break
+		}
+		if strings.Contains(err.Error(), "fingerprint") {
+			continue
+		}
```

Comment:
nit: Would prefer that we use backoff method for consistency. Is 3 tries sufficient?

Reply:
Also should we wait or jitter ?

Reply:
It really only takes one extra try, there's not been a need for a delay or jitter that I've seen

Reply:
Why doesn't returning the error and re-entering the operator loop fix this? In other words, why do we need to manually retry?

Reply:
Because it was running into a conflict error when updating the status and wouldn't actually go into an error state.